src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -335,6 +335,15 @@ public byte[] readAllBytes() throws IOException {
         return (capacity == nread) ? buf : Arrays.copyOf(buf, nread);
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @since 11
+     *
+     * @throws    IllegalArgumentException {@inheritDoc}
+     * @throws    IOException {@inheritDoc}
+     * @throws    OutOfMemoryError {@inheritDoc}
```

Comment:
Re: inheriting _unchecked_ exception documentation here and elsewhere in this PR

This PR's title suggests that the PR has nothing to do with exception documentation. If you feel that it should be addressed, please file a separate bug and move these changes there.

FWIW, I agree that `@throws ... {@inheritDoc}` for such exceptions are at least worth being considered. It's a common misconception that all exception documentation is inherited automatically.

Reply:
This is because the tool from #18934 has no easy way to fetch the doc comment from a superclass/superinterface overridden method when this class only has a plain override. Javadoc handles this in complex logic in `VisibleMemberTable`; it's hard for other clients to try to emulate the behavior of doc finding, and the tool just incorrectly assumes such methods (what I have been talking about before) are reusing `@since` from the class docs.

Reply:
Pavel, can I simply change the PR/issue title to be more descriptive? 
As I want to include the the inherit doc because of the unchecked exceptions, rather than clean this up in a different PR

Reply:
> Pavel, can I simply change the PR/issue title to be more descriptive? As I want to include the the inherit doc because of the unchecked exceptions, rather than clean this up in a different PR

I suggest dropping all the changes that are irrelevant to `@since` from this PR, unless you have PRs with similar mixed changes integrated. As far as I know, `@since` is not a normative part of documentation, whereas `@throws` is. So the latter is a bigger change and would require more scrutiny and more area experts during review.

In the future, we might want to have yet another aid/tool: a "throws checker" that finds all overriding methods that do not declare some of the unchecked exceptions that the methods they override do.